ambry-tools/src/main/java/com.github.ambry.tools/admin/DumpData.java

```diff
@@ -454,394 +428,362 @@ public long dumpActiveBlobsFromIndex(File indexFileToDump, ArrayList<String> blo
   }
 
-  public void dumpReplicaToken(File replicaTokenFile)
-      throws IOException, NoSuchMethodException, InstantiationException, IllegalAccessException, ClassNotFoundException,
-             InvocationTargetException {
-    logOutput("Dumping replica token");
-    DataInputStream stream = new DataInputStream(new FileInputStream(replicaTokenFile));
-    short version = stream.readShort();
-    switch (version) {
-      case 0:
-        int Crc_Size = 8;
-        StoreKeyFactory storeKeyFactory = Utils.getObj("com.github.ambry.commons.BlobIdFactory", map);
-        FindTokenFactory findTokenFactory =
-            Utils.getObj("com.github.ambry.store.StoreFindTokenFactory", storeKeyFactory);
-        while (stream.available() > Crc_Size) {
-          // read partition id
-          PartitionId partitionId = map.getPartitionIdFromStream(stream);
-          // read remote node host name
-          String hostname = Utils.readIntString(stream);
-          // read remote replica path
-          String replicaPath = Utils.readIntString(stream);
-          // read remote port
-          int port = stream.readInt();
-          // read total bytes read from local store
-          long totalBytesReadFromLocalStore = stream.readLong();
-          // read replica token
-          FindToken token = findTokenFactory.getFindToken(stream);
-          logOutput(
-              "partitionId " + partitionId + " hostname " + hostname + " replicaPath " + replicaPath + " port " + port
-                  + " totalBytesReadFromLocalStore " + totalBytesReadFromLocalStore + " token " + token);
-        }
-        logOutput("crc " + stream.readLong());
+  /**
+   * Compares every entry in every index file of a replica with those in the log.
+   * Checks to see if each blob in index is successfully deserializable from the log
+   * @param replicaRootDirectory the root directory of the replica
+   * @param logFile the log file that needs to be parsed
+   * @throws Exception
+   */
+  public void compareReplicaIndexEntriestoLogContent(String replicaRootDirectory, String logFile)
+      throws Exception {
+    if (logFile == null) {
+      log("logFile needs to be set for compareIndexToLog", LogLevel.INFO);
+      System.exit(0);
+    }
+    RandomAccessFile randomAccessFile = new RandomAccessFile(new File(logFile), "r");
+    log("Comparing Index entries to Log ", LogLevel.INFO);
+    File replicaDirectory = new File(replicaRootDirectory);
+    MergedIntervals coveredRangesInLog = new MergedIntervals();
+    for (File indexFile : replicaDirectory.listFiles(new IndexFileNameFilter())) {
+      compareIndexEntriesToLogContent(indexFile, replicaDirectory, randomAccessFile, coveredRangesInLog, false);
+    }
+    logUnCoveredRanges(coveredRangesInLog);
+  }
+
+  /**
+   * Logs uncovered ranges in the log
+   * @param coveredRangesInLog
+   */
+  private void logUnCoveredRanges(MergedIntervals coveredRangesInLog) {
```

Comment:
Yes, but it refers to those ranges in the log that are not covered via entries in the index. So, I have named it so.